### PR TITLE
Implement the job progress bar with ipywidgets

### PR DESCRIPTION
**Pull request by MaxleDrut**
We were previously using the stdout to display the output of a job, no matter the environment. However, this would make it render poorly in notebooks, especially in Google Colab were it was not displayed at all.

This PR introduces a basic ipywidget to render the job status when we detect the code is running from a notebook. We keep for now the stdout channel when we run from a python file.
This detection is inspired from what is done with `tqdm.autonotebook`: https://github.com/tqdm/tqdm/blob/master/tqdm/autonotebook.py

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,21 +30,23 @@ classifiers=[
 urls = {Homepage = "https://github.com/Alice-Bob-SW/qiskit-alice-bob-provider", "Alice & Bob" = "https://alice-bob.com/"}
 requires-python = ">=3.8, <3.12"
 dependencies = [
+    "ipython",
```
Comment (MaxleDrut) [inline pyproject.toml:33]: We must not set a fixed version of ipython, to avoid conflicts with users installing the library that already have their own version.

More importantly, if you install the provider from a running notebook (including on colab), it would fail as it will tell you the ipykernel module can not be changed while running. This can go as far as crashing Colab haha.

File: pyproject.toml [hunk 2]
```diff
@@ -30,21 +30,23 @@ classifiers=[
...
-    # >=0.24.0: required for correct handling of durations
     # <0.45.0: this version introduces many API changes that we need to integrate
     "qiskit>=0.43,<0.45",
     # <0.13.0: qiskit-aer has become irritatingly slow in this release.
     # It especially shows in https://github.com/Alice-Bob-SW/emulation-examples/blob/main/3_error_detection_on_physical_qubits.ipynb
     "qiskit-aer<0.13.0",
     "qiskit-qir-alice-bob-fork==0.3.2rc0",
-    "tenacity",
-    "pydantic>=2.0"
+    "tenacity"
```
Comment (MaxleDrut) [inline pyproject.toml:43]: Just reordering the module to follow the alphabetical name, as a driveby

File: qiskit_alice_bob_provider/remote/job.py [hunk 3]
```diff
@@ -232,7 +232,6 @@ def result(
         status = self.status()
         assert self._last_response is not None
         success = status == JobStatus.DONE
-        print()  # Print a new line to display the next status.
```
Comment (MaxleDrut) [inline qiskit_alice_bob_provider/remote/job.py:235]: This was replaced by the more proper `display_new_line` function, to call at the beginning of the call chain

**Review by alxthm — APPROVED** [review]
